bin/deploy.sh

```diff
@@ -29,39 +26,48 @@ echo
 echo ".........................................."
 echo
 
+# Make sure the SVN temporary folder is removed
+rm -fr $SVNPATH/
+
 # Check version in readme.txt is the same as plugin file after translating both to unix line breaks to work around grep's failure to identify mac line breaks
-PLUGINVERSION=`grep "Version:" $GITPATH/$MAINFILE | awk -F' ' '{print $NF}' | tr -d '\r'`
+PLUGINVERSION=`grep "Version:" $PLUGINDIR/$MAINFILE | awk -F' ' '{print $NF}' | tr -d '\r'`
 echo "$MAINFILE version: $PLUGINVERSION"
-READMEVERSION=`grep "^Stable tag:" $GITPATH/README.txt | awk -F' ' '{print $NF}' | tr -d '\r'`
+READMEVERSION=`grep "^Stable tag:" $PLUGINDIR/README.txt | awk -F' ' '{print $NF}' | tr -d '\r'`
 echo "README.txt version: $READMEVERSION"
 
 # Check the version in package.json and make sure it's also the same
 PACKAGEVERSION=`cat $CURRENTDIR/package.json | jq -r '.version'`
 echo "package.json version: $PACKAGEVERSION"
 
 if [ "$READMEVERSION" = "trunk" ]; then
-	echo "Version in readme.txt & $MAINFILE don't match, but Stable tag is trunk. Let's proceed..."
+	echo "Version in readme.txt & $MAINFILE don't match, but Stable tag is trunk. Continuing..."
 elif [ "$PLUGINVERSION" != "$READMEVERSION" ]; then
 	echo "Version in readme.txt & $MAINFILE don't match. Exiting..."
 	exit 1;
 elif [ "$PACKAGEVERSION" != "$PLUGINVERSION" ]; then
 	echo "Version in package.json & $MAINFILE don't match. Exiting..."
 	exit 1;
 elif [ "$PLUGINVERSION" = "$READMEVERSION" ]; then
-	echo "Versions match in readme.txt, package.json and $MAINFILE. Let's proceed..."
+	echo "Versions match in readme.txt, package.json and $MAINFILE. Continuing..."
 fi
 
-# Temporary so we don't actually push
-echo "Exit before deploying since this is a test."
-exit 1;
-
-echo "Changing to $GITPATH"
-cd $GITPATH
-
 echo
 echo "Creating local copy of SVN repo trunk ..."
-svn checkout $SVNURL $SVNPATH --depth immediates
-svn update --quiet $SVNPATH/trunk --set-depth infinity
+svn checkout $SVNURL $SVNPATH --non-interactive --no-auth-cache --username=$SVNUSER --password=$WP_ORG_PASSWORD --depth immediates
+svn update --quiet $SVNPATH/trunk --set-depth infinity --non-interactive --no-auth-cache --username=$SVNUSER --password=$WP_ORG_PASSWORD
+
+# Check latest version tag on SVN and see if this version is a duplicate
+cd $SVNPATH
+TAGREVISION=`svn info ^/tags/$PLUGINVERSION --non-interactive --no-auth-cache --username=$SVNUSER --password=$WP_ORG_PASSWORD | grep Revision | tr -d 'Revison: '`
+
+if [ -z "$TAGREVISION" ]; then
+    echo "No tag for $PLUGINVERSION yet. Continuing..."
+else
+    echo "Error: A tag exists with the plugin version $PLUGINVERSION. Make sure you update the plugin version before deploying"
+    exit 1;
```

Comment:
In a future revision, we could provide a way of updating the assets and/or readme file without bumping the version so that we can just update our plugin listing page even if no code changes were made.